src/controllers/payments.js

```diff
@@ -140,9 +141,11 @@ exports.put = function *(id) {
 
   payment.id = id.toLowerCase()
 
-  let isPrepared = yield Payments.validate(payment)
-  if (isPrepared) {
-    yield Payments.settle(payment)
+  yield Payments.validate(payment)
+  const truePayment = settlementQueue.storePayment(payment)
```

Comment:
Ok, I think I understand roughly what's going on. It's a bit confusing that `storePayment` returns a new type of "payment". Maybe it would be more clear to split this up into two function calls, one that just stores the payment and returns nothing and one that gets the trusted payment.